packages/@aws-cdk/aws-logs/lib/log-retention-provider/index.ts

```diff
@@ -18,14 +18,35 @@ interface SdkRetryOptions {
  * @param options CloudWatch API SDK options.
  */
 async function createLogGroupSafe(logGroupName: string, region?: string, options?: SdkRetryOptions) {
-  try { // Try to create the log group
-    const cloudwatchlogs = new AWS.CloudWatchLogs({ apiVersion: '2014-03-28', region, ...options });
-    await cloudwatchlogs.createLogGroup({ logGroupName }).promise();
-  } catch (e) {
-    if (e.code !== 'ResourceAlreadyExistsException') {
-      throw e;
+  // If we set the log retention for a lambda, then due to the async nature of
+  // Lambda logging there could be a race condition when the same log group is
+  // already being created by the lambda execution. This can sometime result in
+  // an error "OperationAbortedException: A conflicting operation is currently
+  // in progress...Please try again."
+  // To avoid an error, we do as requested and try again.
+  let retryCount = options?.maxRetries == undefined ? 10 : options.maxRetries;
+  const delay = options?.retryOptions?.base == undefined ? 10 : options.retryOptions.base;
+  do {
+    try {
+      const cloudwatchlogs = new AWS.CloudWatchLogs({ apiVersion: '2014-03-28', region, ...options });
+      await cloudwatchlogs.createLogGroup({ logGroupName }).promise();
+      return;
+    } catch (error) {
+      if (error.code == 'ResourceAlreadyExistsException') {
+        // Yes, the log group is already created by the lambda execution
+        return;
+      }
+      if (error.code != 'OperationAbortedException') {
+        throw error;
+      }
+      if (retryCount == 0) {
+        console.error(error);
+        throw error;
+      }
     }
-  }
+    await new Promise(resolve => setTimeout(resolve, delay));
+  } while (retryCount > 0);
```

Comment:
Looks like an infinite loop. Where is `retryCount` being reduced?

Reply:
Fixed
A unit-test is added to check this.